File: typescript/smily.ts
/**
 *
 * @param texts list of string [":)", ":(", ":>"]
 * @returns amount of smily face detected
 */

export function countSmilyFace(texts: string[]): number {
  const regex = /[:;][-~]?[)D]/g;
  const matching = texts.filter((smile) => smile.match(regex));
  return matching.length;
}

File: golang/smily.go
package main

import "regexp"

func countSmilyFace(texts []string) int {
	regex := "[:;][-~]?[)D]"
	re := regexp.MustCompile(regex)

	count := 0
	for _, text := range texts {
		if re.MatchString(text) {
			count++
		}
	}

	return count
}
